File: 3712-sum-of-elements-with-frequency-divisible-by-k/3712-sum-of-elements-with-frequency-divisible-by-k.js
/**
 * @param {number[]} nums
 * @param {number} k
 * @return {number}
 */
var sumDivisibleByK = function(nums, k) {
    let sum=0
    let obj={}
    for(i=0;i<nums.length;i++){
        value=nums[i]
        if(obj[value]){
            obj[value]++
        }
        else{
            obj[value]=1
        }
    }
    for (let key in obj){
        if(obj[key]%k==0){
            sum+=obj[key]*key
        }
    }
    return sum
};

File: 3712-sum-of-elements-with-frequency-divisible-by-k/3712-sum-of-elements-with-frequency-divisible-by-k.py
class Solution(object):
    def sumDivisibleByK(self, nums, k):
        sum=0
        obj={}
        for i in range(len(nums)):
            value=nums[i]
            if value in obj:
                obj[value]+=1
            else:
                obj[value]=1
        
        
        for key in obj:
            if obj[key]%k==0:
                sum+=obj[key]*key
        return sum
        